hybrasyl/Monolith.cs

```diff
@@ -138,6 +143,155 @@ private static void SpawnMonster(Monster monster, Map map)
+        /// </summary>
+        /// <param name="lootTableToConvert">Hybrasyl.LootTable to Convert</param>
+        /// <returns>Creatures.LootTable</returns>
+        private Creatures.LootTable ConvertLootTable(Hybrasyl.LootTable lootTableToConvert)
+        {
+            var newLootTable = new Creatures.LootTable();
+            newLootTable.Items = new Creatures.LootTableItemList();
+            newLootTable.Items.Items = new List<Creatures.LootItem>();
+
+            newLootTable.Chance = lootTableToConvert.Chance;
+            newLootTable.Gold.Min = lootTableToConvert.Gold.Min;
+            newLootTable.Gold.Max = lootTableToConvert.Gold.Max;
+            newLootTable.Items.Chance = lootTableToConvert.Items.Chance;
+
+            foreach(var items in lootTableToConvert.Items.Items)
+            {
+                var newLootItem = new Creatures.LootItem();
+                newLootItem.Always = items.Always;
+                newLootItem.Max = items.Max;
+                newLootItem.Min = items.Min;
+                newLootItem.Unique = items.Unique;
+                newLootItem.Value = items.Value;
+                newLootItem.Variants = items.Variants;
+
+                newLootTable.Items.Items.Add(newLootItem);
+            }
+            
+            newLootTable.Items.Rolls = lootTableToConvert.Items.Rolls;
+            newLootTable.Rolls = lootTableToConvert.Rolls;
+
+            return newLootTable;
+        }
+
+        /// <summary>
+        /// Calculate the total amount of lootable gold based on the base amount plus any additional from the loot table(s).
+        /// </summary>
+        /// <returns>Gold uint</returns>
+        public uint LootableGold()
+        {
+            uint lootableGold = 0;
+            var spawnLootGold = _spawn.Loot.Gold;
+
+            if (spawnLootGold != null)
+            {
+                lootableGold += ((uint)_rng.Next((int)(spawnLootGold?.Min ?? 0), (int)(spawnLootGold?.Max ?? 0)));
+            }
+            if (_spawn.Loot.Table != null || _spawn.Loot.Set != null)
+            {
+                _spawnLootTable.ForEach(lootTable => lootableGold += ((uint)_rng.Next((int)(lootTable.Gold?.Min ?? 0), (int)(lootTable.Gold?.Max ?? 0))));
+            }
+
+            return lootableGold;
+        }
+
+        /// <summary>
+        /// Creates a list of LootItems from the LootTables
+        /// </summary>
+        /// <returns>List of LootItems</returns>
+        public List<ItemObject> LootableItems()
+        {
+            List<ItemObject> lootableItems = new List<ItemObject>();
+            var worldItems = Game.World.WorldData.Values<Hybrasyl.Items.Item>();
+
+            if (_spawnLootTable.Count > 0)
+            {
+                foreach (var table in _spawnLootTable)
+                {
+                    foreach (var item in table.Items.Items)
+                    {
+                        for (int i = 0; i < table.Items.Rolls; i++)
```

Comment:
I think there may be a little misunderstanding here. You don't roll on each item; you roll to get a random item from the _table itself_. So the roll/chance doesn't impact each item; it impacts the entire table.

For instance, consider the following fragment:

```xml
<Items Rolls="6" Chance="0.25">
  <Item>Stick</Item>
  <Item>Mold</Item>
  <Item>Hot Garbage</Item>
  <Item>Swonko's Letter</Item>
</Items>
```

The intent is that this would result in 6 rolls being made, with a 25% chance to get an item off of the table (per roll). The item selection is random, _unless_ the drop is enforced using `Always`:

`<Item Always="true">Stick</Item>` - would always result in _one_ stick dropping

Also, you need to account for `Unique` (only drop once, regardless of # of rolls / # of hits) and `Variants`. 

`Variants` in particular need to be handled in the first pass to handle woodlands areas.

Reply:
1) Do I have the implementation of the tables correct then, with regard to the rolls? Or should it be possible for the same table to be used potentially multiple times (up to the number of rolls)?

2) Using your example, if Rolls = 1 and `<Item Always="true">Stick</Item>`, the only the stick would ever drop, correct?

3) What exactly are `variants`? Given that they only require a string value, I didn't understand what this was meant to be, so I was going to do this in a second pass.

Reply:
Re: 3

Variants are item variants, such as "Deoch Jade Ring", "Good Battle Sword", "Might Mythril Bracer", etc.  Am uncertain how they are constructed, but defined via xml.

Reply:
Of course. Thanks for clarifying. That part makes sense to me now.

Reply:
Do I need to consider the `variants` of the item itself as well? Or just the `variants` referred to in the loot table? My assumption is the latter, but wanted to confirm.

Reply:
Sorry it took a while for me to respond, was indisposed yesterday.

1) Yes, it is possible for the same table to be used multiple times.
2) "Always" was intended to always be *additive* and only occurs once (so for 4 items plus an always item, you would get the always item and a randomly selected item from the 4).
3) Variants are referred to in the loot table. e.g. stick, variant great would have a chance to drop Great Stick.

Reply:
This is done.